maven-plugin/pom.xml

```diff
@@ -66,9 +66,8 @@
       <artifactId>maven-plugin-annotations</artifactId>
     </dependency>
     <dependency>
-      <groupId>junit</groupId>
-      <artifactId>junit</artifactId>
-      <scope>test</scope>
+      <groupId>xml-apis</groupId>
+      <artifactId>xml-apis</artifactId>
```

Comment:
In this module (maven-plugin) and the start module, the xml-apis jar is being marked as used but undeclared by dep analysis.  It looks like this is a weird classloading issue where we either need to exclude xml-apis or mark it as provided as you mentioned.  I will mark it as provided in both modules and see if that works.